### PR TITLE
RD-6325 Add support for `get_secrets` used as `valid_values` constraint

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added support for get_secrets used as valid_values constraint

## Screenshots / Videos
N/A - uses existing UI

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/4162/

 

## Documentation
N/A - existing documentation does not go into level of underlying implementation details that wold require an update

### CODE REVIEW

**Comment by PatrycjaKubiczek** [comment]
@kubama did you run system test again after the changes you made? it seems like the https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/4145/ shows some errors which seem to be connected to the changes from this PR 🤔

**Comment by kubama** [comment]
@PatrycjaKubiczek The failures are addressed. I updated the description with a link to a re-run.

File: app/widgets/common/inputs/fields/ValueListInputField.tsx [hunk 1]
```diff
@@ -1,35 +1,73 @@
...
+        setOptions(
+            map(values, validValue => {
+                const parsedOptionValue = parseOptionValue(validValue);
+
+                return {
+                    name: validValue,
+                    text: validValue,
+                    value: parsedOptionValue
+                };
+            })
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/widgets/common/inputs/fields/ValueListInputField.tsx]: Done

File: app/widgets/common/inputs/fields/ValueListInputField.tsx [hunk 2]
```diff
@@ -1,35 +1,73 @@
...
+                            optionsToSet = parsedValue;
+                        } else if (isPlainObject(parsedValue)) {
+                            optionsToSet = Object.keys(parsedValue);
+                        } else {
+                            optionsToSet = [parsedValue];
+                        }
+                    } else {
+                        optionsToSet = [secretValue];
+                    }
+                    initOptions(optionsToSet);
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/widgets/common/inputs/fields/ValueListInputField.tsx]: Done

**Review by PatrycjaKubiczek — APPROVED** [review]

**Pull request merged**